### PR TITLE
feat: implement smart text chunking for embed fields

### LINKED ISSUE #75: Enhancement(astral):
### Summary

Embed fields are limited to <1024 characters, so we introduced chunking logic.
https://github.com/JacobCoffee/byte/blob/f814685dd8af28d979b9cc735527f2d367073b93/src/byte/plugins/astral.py#L72-L73

For the most part this does just fine, but it leads to ugly lines in embeds occasionally - and more importantly it can break markdown links provided by the `ruff` output

image

### Basic Example

Make the chunking "smarter" to not break lines at awkward places

### Drawbacks and Impact

_No response_

### Unresolved questions

- how

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Summary
- Resolves #75 - Don't break markdown links in Ruff output
- Adds `smart_chunk_text()` function that respects markdown structures

## Changes
- **New function**: `smart_chunk_text(text, max_size=1000)` in `services/bot/src/byte_bot/lib/utils.py`
  - Preserves markdown links `[text](url)`
  - Preserves inline code `` `code` ``
  - Preserves code blocks ` ```code``` `
  - Prefers split points: paragraphs > sentences > newlines > spaces
- **Updated**: `/ruff` command now uses `smart_chunk_text` instead of naive `chunk_sequence`
- **Added**: 15 comprehensive tests

## Test plan
- [x] Markdown links are not broken mid-chunk
- [x] Code blocks are not broken
- [x] Inline code is not broken
- [x] Chunks don't exceed max_size
- [x] All 1140 tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add smart markdown-aware text chunking for Ruff rule explanations and wire it into the Astral /ruff command.

New Features:
- Introduce smart_chunk_text utility to split text into size-limited chunks while preserving markdown links, inline code, and code blocks.

Enhancements:
- Update Ruff rule embed generation to use smart markdown-aware chunking instead of generic sequence chunking for explanations.

Tests:
- Add comprehensive unit tests covering smart_chunk_text behavior, including markdown preservation, boundary preferences, and size limits.

### CODE REVIEW

File: services/bot/src/byte_bot/lib/utils.py [hunk 1]
```diff
@@ -153,6 +154,87 @@ def chunk_sequence[T](sequence: Iterable[T], size: int) -> Iterable[tuple[T, ...
...
+def smart_chunk_text(text: str, max_size: int = 1000) -> list[str]:
+    """Split text into chunks without breaking markdown structures.
+
+    Respects markdown links, inline code, and code blocks. Prefers splitting at
+    natural boundaries: paragraphs > sentences > newlines > spaces.
+
+    Args:
+        text: The text to chunk.
+        max_size: Maximum characters per chunk.
+
```
[the remark this answers is not in the document]
Comment (JacobCoffee) [reply services/bot/src/byte_bot/lib/utils.py]: Fixed in a558827 - Added validation that raises `ValueError` if `max_size <= 0`.

**Comment by JacobCoffee** [comment]
Addressed the critical issues in a558827:

**1. max_size validation** - Now raises `ValueError` if `max_size <= 0` to prevent infinite loop

**2. Code block test** - Now verifies fences stay in same chunk with matched count assertions

**Regarding the other feedback:**

- **Performance of `_is_position_protected`**: Acceptable for Discord embed text sizes (typically < 6000 chars with few protected regions). Would only optimize if profiling shows it's a bottleneck.

- **Long code blocks > max_size**: The current behavior (allowing oversized chunks) is intentional - it's better to have one large chunk containing the complete code block than to break the markdown. This matches the "best effort" nature of the function.

All 1141 tests pass.
